### ISSUE TITLE
Spurious error when installing nightly build

### ISSUE THREAD

**Issue by athomann** [facebookexperimental/Recoil#1541]
I'm attempting to install the nightly build via yarn but the command fails with this error.

```
error /Users/home/dev/proj/web/node_modules/recoil: Command failed.
Exit code: 1
Command: cd ./node_modules/gen-flow-files && yarn install && yarn build
Arguments: 
```

**Comment by athomann** [comment]
Was able to get around the error with `--ignore-scripts`

**Comment by drarmstr** [comment]
That post-build script is used to generate the Flow files and should already be done as part of the original install when the nightly branch was built.  So, this  error shouldn't  be a problem and ignoring it should be safe.  Would you be up to help adjust `scripts/deploy_nightly_build.js` to clean this up?

Or even better if we could find a better alternative to using/patching gen-flow-files...

**Comment by gurkerl83** [comment]
I think cleaning up might result in removing the gen-flow-files dependency at all, at least how it integrated right now targeting a git repo at the source level, not a bundle after a build similar to what recoils nightly branch does. 

@drarmstr It seems you forked the gen-flow-files repo, made some changes with babel, and pushed the commits. 

My question is about the requirement to use the dependency; respectively, I`m questioning how flow files are spit out by this tool, corresponding to the recoils source folder structure. A way to prevent this may be a flow plugin for the bundler. 

One thing to note is that the generated flow files are distributed for the commonJS format only. See e.g.

https://github.com/facebookexperimental/Recoil/tree/nightly/cjs/core

I don`t have experience with flow files and how to handle them. Thx!

**Comment by drarmstr** [comment]
Yes exactly, the proper cleanup is probably migrating `gen-flow-files` to `flow-copy-source` or something else (cc @mondaychen).  The reason we are using a forked version of `gen-flow-files` is just because it is no longer maintained and we needed an update to support newer syntax with a newer Babel.

**Comment by songning0605** [comment]
> That post-build script is used to generate the Flow files and should already be done as part of the original install when the nightly branch was built. So, this error shouldn't be a problem and ignoring it should be safe. Would you be up to help adjust `scripts/deploy_nightly_build.js` to clean this up?
> 
> Or even better if we could find a better alternative to using/patching gen-flow-files...

**Comment by SmallHollow** [comment]
@drarmstr I get this same error while trying install Recoil 0.6.0 (`npm install recoil@0.6.0` with npm 8.1.2).

**Comment by salvoravida** [comment]
> @drarmstr I get this same error while trying install Recoil 0.6.0 (`npm install recoil@0.6.0` with npm 8.1.2). With `--ignore-scripts` the install works.

same here

**Comment by gurkerl83** [comment]
The error can also be suppressed in yarn. Here is the necessary configuration in package.json

```
"dependenciesMeta": {
   "recoil": {
      "built": false
   }
},
```

Tested with Yarn 3.

If an app is deployed with NextJs / Vercel, presumably other providers are also affected, this configuration must be included, otherwise a deployment will fail.

**Comment by jonbnewman** [comment]
This error is showing up with the latest general release `v0.6.0` @drarmstr

**Comment by salvoravida** [comment]
if it can help, meanwhile, I have published a clone without a "post-install" script, use it with yarn resolutions

```
  "resolutions": {
    "recoil": "yarn:@salvoravida/recoil"
  },
```

**Comment by gurkerl83** [comment]
Due to the large number of downloads (>77k) of the latest stable version 0.5.2, it might be appropriate to temporarily remove the postBuild step and trigger new deployment on NPM.

As described in another ticket, the incorrect build step tries to generate flow files after the download performed by a package manager. 

These are generated like the bundles themselves before a deployment, and shipped on NPM as well. A repeated build with missing sources on the client leads to the error.

**Comment by drarmstr** [comment]
Fixed with #1577 and `0.6.1`
